assignments/sa/redux/index.md

```diff
@@ -379,16 +379,286 @@ git tag v4
+
+### Prepare the Repo
+
+🚀 To begin, create a new branch `redux-sa` in your SA4 repo: https://github.com/dartmouth-cs52-19S/sa4-YOUR_USERNAME
+
+Then navigate to your repository on the command line and pull in the updates you've just made to your starterpack:
+
+``` bash
+cd sa4-YOUR_USERNAME
+git pull
+git checkout redux-sa
+git pull starter master
+```
+Hopefully, you should only have major conflicts in your `src/index.js` file. Let's handle them now!
+
+🚀 First, copy all the code from the HEAD (Current Change) of `src/index.js` and paste it into its own new file `src/components/youtube.js`. Now go ahead and accept the incoming change.
+
+<details>
+<summary>Don't worry if you lost your HEAD, it's right here!</summary>
+
+```javascript
+import React, { Component } from 'react';
+import ReactDOM from 'react-dom';
+import './style.scss';
+import debounce from 'lodash.debounce';
+import SearchBar from './components/search_bar';
+import youtubeSearch from './youtube-api';
+import VideoList from './components/video_list';
+import VideoDetail from './components/video_detail';
+
+class App extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      videos: [],
+      selectedVideo: null,
+    };
+
+    this.search = debounce(this.search, 300);
+    this.search('pixar');
+  }
+
+  search = (text) => {
+    youtubeSearch(text).then((videos) => {
+      this.setState({
+        videos,
+        selectedVideo: videos[0],
+      });
+    });
+  }
+
+  render() {
+    return (
+      <div>
+        <SearchBar onSearchChange={this.search} />
+        <div id="video-section">
+          <VideoDetail video={this.state.selectedVideo} />
+          <VideoList onVideoSelect={(selectedVideo) => this.setState({ selectedVideo })} videos={this.state.videos} />
+        </div>
+      </div>
+    );
+  }
+}
+```
+</details>
+
+🚀 To finish off our merge, let's create a new endpoint in `components/app.js` for YouTube:
+
+```javascript
+// Add this import somewhere at the top
+import YouTube from './youtube';
+
+// Add a new NavLink to your Nav constant
+<li><NavLink to="/youtube" exact>YouTube</NavLink></li>
+
+// And add a new Route to the Switch in App
+<Route exact path="/youtube" component={YouTube} />
+```
+
+🚀 Once you have finished handling the conflicts, go ahead and add, commit, and push the changes.
+
+To update our code, we will follow largely the same process as above, but with more complicated functionality.
+
+### Actions
+
+Since selecting a video is a new action handled by our redux store, we will need to create a new function in `actions/index.js` to go along with it.
+
+🚀 First let's expand our `ActionTypes`:
+
+```javascript
+// keys for actiontypes
+export const ActionTypes = {
+  INCREMENT: 'INCREMENT',
+  DECREMENT: 'DECREMENT',
```

Comment:
Not sure they will need this for the Youtube short, right? Might be confusing to include it

Reply:
My thinking was that it would be more trouble than it's worth to have people yank it all out. Let me know what you think after a run through!